Mathlib/CategoryTheory/Enriched/Limits/IsConicalLimit.lean

```diff
@@ -0,0 +1,152 @@
+  -/
+  isConicalLimit (X : C) : IsLimit <| (eCoyoneda V X).mapCone c
+
+namespace IsConicalLimit
+
+variable {V} {c}
+
+/-- Transport evidence that a cone is a `V`-enriched limit cone across an isomorphism of cones. -/
+noncomputable def ofIso {r₁ r₂ : Cone F} (h : IsConicalLimit V r₁) (i : r₁ ≅ r₂) :
+    IsConicalLimit V r₂ where
+  isLimit := h.isLimit.ofIsoLimit i
+  isConicalLimit X := h.isConicalLimit X |>.ofIsoLimit
+    { hom := Functor.mapConeMorphism _ i.hom
+      inv := Functor.mapConeMorphism _ i.inv
+      hom_inv_id := by simp only [Functor.mapCone, Functor.mapConeMorphism, Iso.map_hom_inv_id]
+      inv_hom_id := by simp only [Functor.mapCone, Functor.mapConeMorphism, Iso.map_inv_hom_id] }
+
+/-!
+## Characterization in terms of the comparison map.
+
+There is a canonical comparison map with the limit in `V`, the following proves that a limit
+cone in `C` is a `V`-enriched limit if and only if the comparison map is an isomorphism
+for every `X : C`.
+-/
+
+-- Adjusting the size of `J` would also work, but this is more universe polymorphic.
+variable [HasLimitsOfShape J V]
+
+variable (V) (c) in
+
+/-- The canonical comparison map with the limit in `V`. -/
+noncomputable def limitComparison : (X ⟶[V] c.pt) ⟶ limit (F ⋙ eCoyoneda V X) :=
+  limit.lift _ <| (eCoyoneda V X).mapCone c
+
+lemma limitComparison_eq_conePointUniqueUpToIso (h : IsConicalLimit V c)
+    [HasLimit (F ⋙ eCoyoneda V X)] :
+    limitComparison V c X =
+    ((h.isConicalLimit X).conePointUniqueUpToIso (limit.isLimit _)).hom := by
+  apply limit.hom_ext
+  simp [limitComparison]
+
+/-- `IsConicalLimit.limitComparison` is an isomorphism. -/
+lemma isIso_limitComparison (h : IsConicalLimit V c) : IsIso (limitComparison V c X) := by
+  rw [limitComparison_eq_conePointUniqueUpToIso (h := h)]
+  infer_instance
+
+/-- For all `X : C`, the canonical comparison map with the limit in `V` as isomorphism -/
+noncomputable def limitComparisonIso (h : IsConicalLimit V c) :
+    (X ⟶[V] c.pt) ≅ (limit (F ⋙ eCoyoneda V X)) := by
+  have := isIso_limitComparison X h
+  exact (asIso (limitComparison V c X))
+
+variable (V) in
+
+/-- Reverse direction: if the comparison map is an isomorphism, then `c` is a conical limit. -/
+noncomputable def ofIsIsoLimitComparison
+    [∀ X, IsIso (IsConicalLimit.limitComparison V c X)]
+    (hc : IsLimit c) : IsConicalLimit V c where
+  isLimit := hc
+  isConicalLimit X := by
+    suffices PreservesLimit F (eCoyoneda V X) from
+      Classical.choice (this.preserves hc)
+    have : HasLimit F := ⟨c, hc⟩
+    apply (config := { allowSynthFailures := true }) preservesLimit_of_isIso_post
+    have h : limit.post F (eCoyoneda V X) =
+      ((eCoyoneda V X).map ((limit.isLimit F).conePointUniqueUpToIso hc).hom) ≫
+        limitComparison V c X := by
+      apply limit.hom_ext
+      intro j
+      simp [limitComparison, ← eHomWhiskerLeft_comp]
+    rw [h]
+    infer_instance
```

Comment:
```suggestion
    suffices limit.post F (eCoyoneda V X) =
      ((eCoyoneda V X).map ((limit.isLimit F).conePointUniqueUpToIso hc).hom) ≫
        limitComparison V c X from this ▸ inferInstance
    apply limit.hom_ext
    simp [limitComparison, ← eHomWhiskerLeft_comp]
```
Others might have the opposite preference, though.

Reply:
took almost this but used 
` by rw [h]; infer_instance` instead of `this ▸ inferInstance`